### PR TITLE
[kube-prometheus-stack] make common part of runbook_url overridable

### PR DESCRIPTION
**Pull request by hairmare**
#### What this PR does / why we need it:

Make it possible to override the common part of the `runbook_url`s in the provided prometheus rules. The upstream mixin allows doing this and it would be nice to have the same feature available to users of `kube-prometheus-stack`.

#### Special notes for your reviewer:

I decided to default the runbookUrlPrefix in the individual yaml files so i could keep the magic `https://runbooks.prometheus-operator.dev/runbooks/` string in one place but i'd be happy to move the default to values.yaml if requested to do so.

I also considered using some `helm template | kustomize` magic to acheive the same results. This change feels like it has less tradeoffs.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by mrueg** [comment]
If folks are using it and willing to support it, let's add it back. :) Can you add a few additional words to the changelog https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md#from-24x-to-25x ?

**Comment by hairmare** [comment]
i changed it to use the same value that was previously used and will update the changelog towmorrow.

File: charts/kube-prometheus-stack/values.yaml [hunk 1]
```diff
@@ -69,6 +69,9 @@ defaultRules:
   ## Additional labels for PrometheusRule alerts
   additionalRuleLabels: {}
 
+  ## Prefix for runbook URLs
+  runbookUrl: ""
```
Comment (eyenx) [inline charts/kube-prometheus-stack/values.yaml:73]: As it looks like, the runbookUrl needs to end with a trailing slash. Should we add this to the comment above?

Comment (Xtigyro) [reply charts/kube-prometheus-stack/values.yaml:73]: yeah - I think we should.

Comment (hairmare) [reply charts/kube-prometheus-stack/values.yaml:73]: The new default runbookUrl needs to end in a trailing slash. If you override it there are a lot of possibilities some of which would not need to end in a trailing slash. The following would both build perfectly valid URLs:

* `https://example.com/path/to/endpoint?alert=`
* `https://example.com/path/to/html-page#`

**Comment by hairmare** [comment]
I added info to the changelog, reworded the comment in `values.yaml` and rebased this PR.

**Review by monotek — CHANGES_REQUESTED** [review]
Why is the trailing slash not just added to the template?

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 25.1.0
+version: 25.2.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:20]:
```suggestion
version: 25.3.0
```

**Comment by monotek** [comment]
@hairmare could yuo please rebase / fix the merge conflict?
I cant push to your branch...

**Comment by monotek** [comment]
superseeded by: https://github.com/prometheus-community/helm-charts/pull/1646

**Pull request closed without merging**